astarte/device/interface.py

```diff
@@ -74,36 +79,57 @@ def __init__(self, interface_definition: dict):
         ValueError
             if both version_major and version_minor numbers are set to 0
         """
+
         self.name: str = interface_definition.get("interface_name")
+        if not isinstance(self.name, str):
+            raise InterfaceFileDecodeError(
+                "Interface name is a required interface field and should be a string."
+            )
+        if name_regex.match(self.name) is None:
+            raise InterfaceFileDecodeError(
+                f"Interface name is not correctly formatted: {self.name}"
+            )
+
         self.version_major: int = interface_definition.get("version_major")
+        if not isinstance(self.version_major, int):
+            raise InterfaceFileDecodeError(
+                "Major version is a required interface field and should be an integer."
+            )
+
         self.version_minor: int = interface_definition.get("version_minor")
-        self.type: str = interface_definition.get("type")
-        self.ownership = interface_definition.get("ownership")
-
-        if not (
-            isinstance(self.name, str)
-            and isinstance(self.version_major, int)
-            and isinstance(self.version_minor, int)
-            and self.type in {"datastream", "properties"}
-            and self.ownership in (DEVICE, SERVER)
-        ):
+        if not isinstance(self.version_minor, int):
             raise InterfaceFileDecodeError(
-                f"Error parsing the following interface definition: {interface_definition}"
+                "Minor version is a required interface field and should be an integer."
             )
 
-        if not self.version_major and not self.version_minor:
+        if (not self.version_major) and (not self.version_minor):
```

Comment:
`pylint` would be unhappy about that. As we enabled a rule to avoid comparisons to zero.
In this case, it's also not necessary as both versions are checked to be integers a couple of lines above.
But we could consider removing the warning for more complex scenarios, as `==0` is surely more indicative.

Reply:
it's fine anyway, for now we can keep it as is.